client/transport/src/main/java/org/elasticsearch/transport/client/PreBuiltTransportClient.java

```diff
@@ -79,13 +82,24 @@ public PreBuiltTransportClient(Settings settings, Collection<Class<? extends Plu
 
         @Override
         public Settings additionalSettings() {
-            return Settings.builder()
-                    .put(NetworkModule.TRANSPORT_TYPE_KEY, Netty4Plugin.NETTY_TRANSPORT_NAME)
-                    .put(NetworkModule.HTTP_TYPE_KEY, Netty4Plugin.NETTY_HTTP_TRANSPORT_NAME)
-                    .put("netty.assert.buglevel", true)
+            return Settings.builder().put("netty.assert.buglevel", true)
                     .build();
         }
 
     }
 
+    @Override
+    public void close() {
+        super.close();
+        if (!NetworkModule.TRANSPORT_TYPE_SETTING.exists(settings)
+            || NetworkModule.TRANSPORT_TYPE_SETTING.get(settings).equals(Netty4Plugin.NETTY_TRANSPORT_NAME)) {
+            try {
+                GlobalEventExecutor.INSTANCE.awaitInactivity(5, TimeUnit.SECONDS);
```

Comment:
is it ok to directly throw an exception in case of an interrupted exception in the first call without trying to close the `ThreadDeathWatcher`?